### ISSUE TITLE
Lint/Syntax error on valid Ruby 2.7.7 syntax

### ISSUE THREAD

**Issue by john-h-k** [rubocop/rubocop#11660]
When running RuboCop 1.47 in an RSpec test file (with `rubocop-rails` and `rubocop-rspec` plugins enabled), valid syntax causes a RuboCop error.

```rb
expect(response.body).to eq { success: false, error: "Example" }.to_json

# causes

E: Lint/Syntax: cannot assign to a keyword
(Using Ruby 2.7 parser; configure using TargetRubyVersion parameter, under AllCops)
      expect(response.body).to eq { success: false, error: "Validation failed: Practice group must exist" }.to_json
```

--------

## Expected behavior

To accept the valid syntax

## Actual behavior

It rejects it with the error above

```sh
For /app: configuration from /app/.rubocop.yml
configuration from /bundle/ruby/2.7.0/gems/rubocop-rails-2.18.0/config/default.yml
configuration from /bundle/ruby/2.7.0/gems/rubocop-rails-2.18.0/config/default.yml
Default configuration from /bundle/ruby/2.7.0/gems/rubocop-1.47.0/config/default.yml
configuration from /bundle/ruby/2.7.0/gems/rubocop-capybara-2.17.0/config/default.yml
configuration from /bundle/ruby/2.7.0/gems/rubocop-capybara-2.17.0/lib/../config/default.yml
configuration from /bundle/ruby/2.7.0/gems/rubocop-rspec-2.18.1/config/default.yml
configuration from /bundle/ruby/2.7.0/gems/rubocop-rspec-2.18.1/config/default.yml
Inheriting configuration from /app/.rubocop_todo.yml
.rubocop.yml: Layout/LineLength:Max overrides the same parameter in .rubocop_todo.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /app/test.rb
E

Offenses:

test.rb:3:38: E: Lint/Syntax: unexpected token tCOLON
(Using Ruby 2.7 parser; configure using TargetRubyVersion parameter, under AllCops)
expect(response.body).to eq { success: false, error: "Example" }.to_json
                                     ^
test.rb:3:40: E: Lint/Syntax: cannot assign to a keyword
(Using Ruby 2.7 parser; configure using TargetRubyVersion parameter, under AllCops)
expect(response.body).to eq { success: false, error: "Example" }.to_json
                                       ^^^^^

1 file inspected, 2 offenses detected
Finished in 0.3165558750042692 seconds
```

## Steps to reproduce the problem

* Create a file with any name containing this code

```rb
expect(response.body).to eq { success: false, error: "Example" }.to_json
```

* Run RuboCop on it

## RuboCop version

```
root@3056a63103b4:/app# bundle exec rubocop -V
1.47.0 (using Parser 3.2.1.0, rubocop-ast 1.27.0, running on ruby 2.7.7) [aarch64-linux]
  - rubocop-rails 2.18.0
  - rubocop-rspec 2.18.1
```

**Comment by fatkodima** [comment]
I do not think this is a valid syntax - that hash `{ ... }` is probably parsed as a block, and `success: false` is an invalid syntax within it. You should wrap that hash by parentheses (`()`).

Is it working in other versions of ruby?

**Comment by koic** [comment]
Yeah, this is invalid syntax:

```console
$ ruby -vce 'expect(response.body).to eq { success: false, error: "Example" }.to_json'
ruby 2.7.7p221 (2022-11-24 revision 168ec2b1e5) [x86_64-darwin19]
-e:1: syntax error, unexpected ':', expecting '}'
...response.body).to eq { success: false, error: "Example" }.to...
-e:1: Can't assign to false
...se.body).to eq { success: false, error: "Example" }.to_json
-e:1: warning: possibly useless use of a literal in void context
-e:1: syntax error, unexpected '}', expecting end-of-input
...cess: false, error: "Example" }.to_json
```

```console
$ ruby -vce 'expect(response.body).to eq { success: false, error: "Example" }.to_json'
ruby 3.2.1 (2023-02-08 revision 31819e82c8) [x86_64-darwin19]
-e:1: syntax error, unexpected ':', expecting '}'
...response.body).to eq { success: false, error: "Example" }.to...
-e: compile error (SyntaxError)
```
